### DISCUSSION TITLE
Parametrize master branch to $default-branch

### DISCUSSION THREAD

**Discussion by teutat3s** [upptime/upptime#241] [Ideas] [answered]
Thank you for making this wonderful monitoring tool!

It would be great if the default branch wouldn't need to be hard-coded to `master` but could rather be parametrized during action runtime.
E.g. here: https://github.com/upptime/uptime-monitor/blob/d2fddfb23076b157f9e8d8f642e7002bb97667a1/src/summary.ts#L102

That way if someone were to choose the `main` branch as default branch, links like https://github.com/upptime/upptime/commits/master/history/google.yml could still work and not turn into `404`'s.

I don't know if this is possible though, because it seems the actions get *released* with that default branch name hard-coded, and I didn't find a way to set the current `$default-branch` in the actions source-code during running workflows yet.

Maybe someone else has got a solution to this idea?

**Comment by AnandChowdhary** [comment, marked as answer]
I've just pushed a fix for this. We now use `HEAD` instead of `master`, meaning that it will use the default branch. For example, https://github.com/upptime/upptime/commits/HEAD/history/google.yml redirects to https://github.com/upptime/upptime/commits/master/history/google.yml.
